### PR TITLE
Add tag sort, filter, search and pagination

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add a segmented button component.
- Add the action bar to Tags to sort, filter and search.
- Add pagination to the table.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit the tags tab under machines.
- Check that you can filter by automatic or manual tags.
- Check that you can search tags.
- Check that you can change the order of the name and updated columns.
- Check that you can paginate the table (you may have to reduce the `TAGS_PER_PAGE` value to less than the number of tags.

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/703.
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/698.
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/697.
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/696.

## Screenshots
Screen Shot 2022-03-04 at 12 07 35 pm

### CODE REVIEW

**Comment by huwshimi** [comment]
Blocked on https://github.com/canonical-web-and-design/maas-ui/pull/3659

File: ui/src/app/base/components/SegmentToggle/SegmentToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          className={classNames("p-segment-toggle__button", {
+            "is-active": value === selected,
+          })}
+          key={title}
+          onClick={() => onSelect(value)}
+        >
+          {title}
+        </Button>
+      ))}
+    </div>
```
Comment (petermakowski) [inline ui/src/app/base/components/SegmentToggle/SegmentToggle.tsx:33]: We already have a Tabs component in Vanilla that seems to do the same thing https://vanillaframework.io/docs/patterns/tabs#default

Could you use the vanilla copmonent instead and follow accessibility recommendations (add roles `tablist`, `tab`, `aria-selected` to the currently active tab).

More on this here:
 https://vanillaframework.io/docs/patterns/tabs#accessibility

File: ui/src/app/tags/views/TagList/TagList.test.tsx [hunk 2]
```diff
@@ -42,120 +39,6 @@ it("displays tags", () => {
...
-  expect(controllerLink).toHaveAttribute(
-    "href",
-    `${controllerURLs.controllers.index}?tags=a-tag`
-  );
-  expect(deviceLink).toHaveAttribute(
-    "href",
-    `${deviceURLs.devices.index}?tags=a-tag`
-  );
+  expect(screen.getByTestId(TestId.TagListControls)).toBeInTheDocument();
+  expect(screen.getByTestId(TestId.TagTable)).toBeInTheDocument();
```
Comment (petermakowski) [inline ui/src/app/tags/views/TagList/TagList.test.tsx:43]: Once you use semantic HTML markup, using a test id will be no longer necessary. Please use a combination of semantic role and label instead.

File: ui/src/app/base/components/ActionBar/ActionBar.tsx [hunk 3]
```diff
@@ -13,37 +11,40 @@ type Props = {
...
+  pageSize = 50,
   searchFilter,
   setCurrentPage,
+  ...props
 }: Props): JSX.Element | null => {
   return (
-    <div className="kvm-action-bar">
-      <div className="kvm-action-bar__actions">{actions}</div>
-      <div className="kvm-action-bar__search">
+    <div className="action-bar" {...props}>
```
Comment (petermakowski) [inline ui/src/app/base/components/ActionBar/ActionBar.tsx:31]:
```suggestion
    <div className="action-bar" role="group" {...props}>
```

File: ui/src/app/tags/views/TagList/TagList.tsx [hunk 4]
```diff
@@ -1,175 +1,50 @@
...
-      ]}
-      rows={generateRows(tags)}
-      paginate={50}
-      sortable
-      defaultSort="name"
-      defaultSortDirection="ascending"
-    />
+    <>
+      <TagListControls
+        data-testid="TagListControls"
```
Comment (petermakowski) [inline ui/src/app/tags/views/TagList/TagList.tsx:30]:
```suggestion
        aria-label="tag list controls"
```

File: ui/src/app/tags/views/TagList/TagList.tsx [hunk 5]
```diff
@@ -1,175 +1,50 @@
...
+        filter={filter}
+        currentPage={currentPage}
+        setFilter={setFilter}
+        searchText={searchText}
+        setCurrentPage={setCurrentPage}
+        setSearchText={setSearchText}
+        tagCount={tags.length}
+      />
+      <TagTable
+        data-testid="TagTable"
```
Comment (petermakowski) [inline ui/src/app/tags/views/TagList/TagList.tsx:40]:
```suggestion
        aria-label="tags"
```

File: ui/src/app/base/components/SegmentToggle/SegmentToggle.tsx (hunk 1 above)
Comment (petermakowski) [reply ui/src/app/base/components/SegmentToggle/SegmentToggle.tsx]: Sure, that's even better. Thanks!

**Review by petermakowski — APPROVED** [review]

**Pull request merged**